apps/web/src/pages/Users/UpdateUserPage/components/IndividualRoleTable.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useMemo } from "react";
+import { useIntl } from "react-intl";
+
+import { notEmpty } from "@gc-digital-talent/helpers";
+import { Pill } from "@gc-digital-talent/ui";
+import { getLocalizedName } from "@gc-digital-talent/i18n";
+
+import Table, { ColumnsOf, Cell } from "~/components/Table/ClientManagedTable";
+import { getFullNameHtml } from "~/utils/nameUtils";
+import { Role, UpdateUserAsAdminInput, User } from "~/api/generated";
+
+import AddIndividualRoleDialog from "./AddIndividualRoleDialog";
+import RemoveIndividualRoleDialog from "./RemoveIndividualRoleDialog";
+import { UpdateUserFunc } from "../types";
+
+const roleCell = (displayName: string) => {
+  return (
+    <Pill color="neutral" mode="solid">
+      {displayName}
+    </Pill>
+  );
+};
+
+const actionCell = (role: Role, user: User, onUpdateUser: UpdateUserFunc) => (
+  <RemoveIndividualRoleDialog
+    role={role}
+    user={user}
+    onUpdateUser={onUpdateUser}
+  />
+);
+
+type RoleCell = Cell<Role>;
+
+interface IndividualRoleTableProps {
+  user: User;
+  availableRoles: Array<Role>;
+  onUpdateUser: UpdateUserFunc;
+}
+
+export const IndividualRoleTable = ({
+  user,
+  availableRoles,
+  onUpdateUser,
+}: IndividualRoleTableProps) => {
+  const intl = useIntl();
+  const userName = getFullNameHtml(user.firstName, user.lastName, intl);
+
+  const columns = useMemo<ColumnsOf<Role>>(
+    () => [
+      {
+        Header: intl.formatMessage({
+          defaultMessage: "Actions",
+          id: "S8ra2P",
+          description: "Title displayed for the role table actions column",
+        }),
+        accessor: (d) => `Actions ${d.id}`,
+        disableGlobalFilter: true,
+        Cell: ({ row: { original: role } }: RoleCell) =>
+          actionCell(role, user, onUpdateUser),
+      },
+      {
+        Header: intl.formatMessage({
+          defaultMessage: "Role",
+          id: "uBmoxQ",
+          description: "Title displayed for the role table display name column",
+        }),
+        accessor: (role) => getLocalizedName(role.displayName, intl),
+        Cell: ({ row: { original: role } }: RoleCell) =>
+          roleCell(getLocalizedName(role.displayName, intl)),
+      },
+    ],
+    [intl, onUpdateUser, user],
+  );
+
+  // TO DO: Update to user roles
+  const data = useMemo(() => {
+    const roles = user.roleAssignments
+      ?.filter(notEmpty)
+      .map((assignment) => assignment.role)
+      .filter(notEmpty);
+    return roles || [];
+  }, [user.roleAssignments]);
+
+  const handleAddRoles = async (values: UpdateUserAsAdminInput) => {
+    return onUpdateUser(user.id, values);
+  };
+
+  return (
+    <Table
+      data={data}
+      columns={columns}
+      addDialog={
+        <AddIndividualRoleDialog
+          userName={userName}
+          availableRoles={availableRoles}
+          onAddRoles={handleAddRoles}
```

Comment:
need to filter out team-based roles somewhere, before passing into the individual role dialog

![image](https://user-images.githubusercontent.com/40485260/222853186-041e6f6b-7438-40ab-bf35-9cde8d36eaa5.png)

Reply:
Filtered out team based roles in [filter out team based roles and current user roles](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5799/commits/0f56ac40106f72881e701ae9633a5a64a41ccb4e)